system-tests/test/deposit-redemption.test.ts

```diff
@@ -372,4 +374,276 @@ describe("System Test - Deposit and redemption", () => {
+
+    it("should reveal the deposit to the bridge", async () => {
+      const { revealedAt } = await TBTC.getRevealedDeposit(
+        depositUtxo,
+        maintainerBridgeHandle
+      )
+      expect(revealedAt).to.be.greaterThan(0)
+    })
+
+    context("when deposit is swept and sweep proof submitted", () => {
+      before("sweep the deposit and submit sweep proof", async () => {
+        ;({ newMainUtxo: sweepUtxo } = await submitDepositSweepTransaction(
+          electrumClient,
+          depositSweepTxFee,
+          systemTestsContext.walletBitcoinKeyPair.wif,
+          true,
+          [depositUtxo],
+          [deposit]
+        ))
+
+        console.log(`
+        Deposit swept on Bitcoin chain:
+        - Transaction hash: ${sweepUtxo.transactionHash}
+      `)
+
+        // Unlike in the deposit transaction case, we must wait for the sweep
+        // transaction to have an enough number of confirmations. This is
+        // because the bridge performs the SPV proof of that transaction.
+        await waitTransactionConfirmed(
+          electrumClient,
+          sweepUtxo.transactionHash
+        )
+
+        await fakeRelayDifficulty(
+          relay,
+          electrumClient,
+          sweepUtxo.transactionHash
+        )
+
+        // TODO: Consider fetching the current wallet main UTXO and passing it
+        //       here. This will allow running this test scenario multiple
+        //       times for the same wallet.
+        await SpvMaintainer.submitDepositSweepProof(
+          sweepUtxo.transactionHash,
+          // This is the first sweep of the given wallet so there is no main UTXO.
+          {
+            // The function expects an unprefixed hash.
+            transactionHash: BitcoinTransactionHash.from(constants.HashZero),
+            outputIndex: 0,
+            value: BigNumber.from(0),
+          },
+          maintainerBridgeHandle,
+          electrumClient
+        )
+
+        console.log(`
+        Deposit sweep proved on the bridge
+      `)
+      })
+
+      it("should broadcast the sweep transaction on the Bitcoin network", async () => {
+        expect(
+          (await electrumClient.getRawTransaction(sweepUtxo.transactionHash))
+            .transactionHex.length
+        ).to.be.greaterThan(0)
+      })
+
+      it("should sweep the deposit on the bridge", async () => {
+        const { sweptAt } = await TBTC.getRevealedDeposit(
+          depositUtxo,
+          maintainerBridgeHandle
+        )
+        expect(sweptAt).to.be.greaterThan(0)
+      })
+
+      it("should increase depositor's balance in the bank", async () => {
+        const { treasuryFee } = await TBTC.getRevealedDeposit(
+          depositUtxo,
+          maintainerBridgeHandle
+        )
+
+        const expectedBalance = depositAmount
+          .sub(treasuryFee)
+          .sub(depositSweepTxFee)
+
+        const actualBalance = await bank.balanceOf(
+          systemTestsContext.depositor.address
+        )
```

Comment:
Same as above.

Reply:
I removed this piece of code.
Since the reveal is done with the vault address specified, the depositor will end up with TBTC tokens minted.
Therefore, I added a check for TBTC tokens balance and not balance in the Bank.